makefiles/tools/openocd-adapters/stlink.inc.mk

```
@@ -1,7 +1,7 @@
 # ST-Link debug adapter
 
 # OpenOCD uses a single configuration file for all ST-Link adapters.
-OPENOCD_ADAPTER_INIT ?= -c 'source [find interface/stlink.cfg]'
+OPENOCD_ADAPTER_INIT ?= -c 'try { source [find interface/stlink.cfg] } on error {} { puts "Warning using old version of OpenOCD" ; source [find interface/stlink-v$(STLINK_VERSION).cfg] } '
```

Why not instead add in `dist/tools/openocd/adapters` a `stlink.cfg` that would hold this more intricate code:

```suggestion
OPENOCD_ADAPTER_INIT ?= -f '$(RIOTBASE)/dist/tools/openocd/adapters/stlink.cfg'
```